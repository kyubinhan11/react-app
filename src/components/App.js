import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route, Switch, Redirect, BrowserRouter } from 'react-router-dom';

import '../styles/App.css';
import * as Routes from '../constants/routes';

import PublicRoute from './routes/PublicRoute';
import PrivateRoute from './routes/PrivateRoute';
import Login from './auths/Login';
import Home from './Home';
import Toast from './Toast';
import PageNotFound from './PageNotFound';

import Auth from '../handlers/auth';
import { logout } from '../actions/authActions';

export class App extends Component {
  componentDidMount() {
    Auth.registerAxiosInterceptor(this.props.logout);
  }

  render() {
    const { user } = this.props;
    
    return (
      <div>
        <BrowserRouter>
          <Switch>
            
            <PublicRoute path={Routes.LOGIN} component={Login} user={user} />
            <PrivateRoute path={Routes.HOME} component={Home} user={user} />

            
            {/* An example of nested routes 
              https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#serving-apps-with-client-side-routing
              <PrivateRoute path={`${match.url}/:topicId`} component={Topic} user={user}/>
              <PrivateRoute exact path={match.url} component={TopicPage} user={user}/>
            */}
            
            <Route path='/' exact render={() => (<Redirect to={Routes.LOGIN}/>)} />
            <Route path='*' exact component={PageNotFound} />

          </Switch>
        </BrowserRouter>
        <Toast />
      </div>  
    );
  }
}

const mapStateToProps = state => {
  const { user } = state.authReducer;
  
  return {
    user
  };  
};

export default connect (
  mapStateToProps, { logout }
)(App)
