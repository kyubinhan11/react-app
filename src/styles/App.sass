@import 'styles/mixins.scss'

body 
  margin: 0
  padding: 0
  font-family: sans-serif

.page-not-found 
  display: flex
  justify-content: center
  align-items: center
  height: 100vh

  +element('title') 
    text-align: center
    font-size: 3rem
    margin-bottom: 20px

  +element('content') 
    font-size: 1.2rem

  +element('link')
    margin: 10px auto
    width: fit-content
    font-size: 1.2rem
  
    +modifier('bolder') 
      border: 1px solid  
    
  


