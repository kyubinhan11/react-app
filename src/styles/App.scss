@import 'styles/mixins.scss';

body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

// block
.page-not-found {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;

  // element
  &__title {
    text-align: center;
    font-size: 3rem;
    margin-bottom: 20px;
  }

  &__content {
    font-size: 1.2rem;
  }

  &__link {
    margin: 10px auto;
    width: fit-content;
    font-size: 1.2rem;

    // modifier
    &--boldered {
      border: 1px solid  
    }
  }
}