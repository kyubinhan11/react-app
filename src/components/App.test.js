import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { App } from './App';

describe('App', () => {
  const app = shallow(<App />);

  it('renders properly', () => {
    expect(app).toMatchSnapshot();
  });
});