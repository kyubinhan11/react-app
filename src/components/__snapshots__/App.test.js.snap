// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BrowserRouter>
          <Switch>
                    <PublicRoute
                              component={[Function]}
                              path="/login"
                              user={undefined}
                    />
                    <PrivateRoute
                              component={[Function]}
                              path="/home"
                              user={undefined}
                    />
                    <Route
                              exact={true}
                              path="/"
                              render={[Function]}
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="*"
                    />
          </Switch>
</BrowserRouter>,
        <Connect(Toast) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <PublicRoute
                        component={[Function]}
                        path="/login"
                        user={undefined}
            />
            <PrivateRoute
                        component={[Function]}
                        path="/home"
                        user={undefined}
            />
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="*"
            />
</Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PublicRoute
                component={[Function]}
                path="/login"
                user={undefined}
/>,
              <PrivateRoute
                component={[Function]}
                path="/home"
                user={undefined}
/>,
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="*"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "path": "/login",
                "user": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "path": "/home",
                "user": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "*",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BrowserRouter>
            <Switch>
                        <PublicRoute
                                    component={[Function]}
                                    path="/login"
                                    user={undefined}
                        />
                        <PrivateRoute
                                    component={[Function]}
                                    path="/home"
                                    user={undefined}
                        />
                        <Route
                                    exact={true}
                                    path="/"
                                    render={[Function]}
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="*"
                        />
            </Switch>
</BrowserRouter>,
          <Connect(Toast) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <PublicRoute
                            component={[Function]}
                            path="/login"
                            user={undefined}
              />
              <PrivateRoute
                            component={[Function]}
                            path="/home"
                            user={undefined}
              />
              <Route
                            exact={true}
                            path="/"
                            render={[Function]}
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="*"
              />
</Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <PublicRoute
                  component={[Function]}
                  path="/login"
                  user={undefined}
/>,
                <PrivateRoute
                  component={[Function]}
                  path="/home"
                  user={undefined}
/>,
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="*"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "path": "/login",
                  "user": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "path": "/home",
                  "user": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "*",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
